service/worker/workerdeployment/workflow.go

```diff
@@ -667,6 +654,22 @@ func (d *WorkflowRunner) handleSetCurrent(ctx workflow.Context, args *deployment
 		d.setDrainageStatus(prevCurrentVersion, enumspb.VERSION_DRAINAGE_STATUS_DRAINING, updateTime)
 	}
 
+	//nolint:staticcheck // deprecated stuff will be cleaned
+	if newCurrentVersion == worker_versioning.UnversionedVersionId && d.State.RoutingConfig.RampingVersion == worker_versioning.UnversionedVersionId {
```

Comment:
moving this _after_ the other version becomes current, otherwise, there would be a small discontinuity.

Reply:
could this cause an NDE because the order of the `syncVersion` events could be flipped in some cases?

Reply:
Thinking about it more, it would only cause an NDE if a user has previously ramped to unversioned and then set current to unversioned. I think it is very unlikely that a pre-release user has done that, so I'm fine with introducing this change without patching